internal/handlers/handler_sign_duo.go

```diff
@@ -25,43 +31,49 @@ func SecondFactorDuoPost(duoAPI duo.API) middlewares.RequestHandler {
 		userSession := ctx.GetSession()
 		remoteIP := ctx.RemoteIP().String()
 
-		ctx.Logger.Debugf("Starting Duo Push Auth Attempt for user '%s' with IP '%s'", userSession.Username, remoteIP)
-
-		values := url.Values{}
+		duoDevice, err := ctx.Providers.StorageProvider.LoadPreferredDUODevice(ctx, userSession.Username)
```

Comment:
```suggestion
		duoDevice, err := ctx.Providers.StorageProvider.LoadPreferredDuoDevice(ctx, userSession.Username)
```

Since this isn't an acronym I think it should be adjusted.